plutus-contract/test/Spec/TxConstraints/MustMint.hs

```diff
@@ -94,6 +106,55 @@ mustMintCurrencyContract = do
     ledgerTx1 <- submitTxConstraintsWith @UnitTest lookups1 tx1
     awaitTxConfirmed $ Tx.getCardanoTxId ledgerTx1
 
+get6 :: [a] -> (a, a, a, a, a, a)
+get6 (a:b:c:d:e:f:_) = (a, b, c, d, e, f)
+get6 _               = error "Spec.Contract.TxConstraints.MustMint.get3: not enough inputs"
+
+{-# INLINABLE mkMustReferenceOutputV2Validator #-}
+mkMustReferenceOutputV2Validator :: TxOutRef -> () -> PV2.ScriptContext -> Bool
+mkMustReferenceOutputV2Validator txOutRef _ =
+    TCV2.checkScriptContext @Void @Void (TC.mustReferenceOutput txOutRef)
+
+{-# INLINABLE mustReferenceOutputV2Validator #-}
+mustReferenceOutputV2Validator :: PV2.Validator
+mustReferenceOutputV2Validator = PV2.mkValidatorScript
+    $$(PlutusTx.compile [|| wrap ||])
+ where
+     wrap = PV2.mkUntypedValidator mkMustReferenceOutputV2Validator
+
+mustReferenceOutputV2ValidatorAddress :: Address
+mustReferenceOutputV2ValidatorAddress =
+    PV2.mkValidatorAddress mustReferenceOutputV2Validator
+
+mustMintValueWithReferenceContract :: Contract () Empty ContractError ()
+mustMintValueWithReferenceContract = do
+    utxos <- ownUtxos
+    myAddr <- Con.ownAddress
+    let (_, _, _, _, _, (utxoRef, utxo)) = get6 $ Map.toList utxos
+        MintingPolicyHash mph = coinMintingPolicyHashV2
+        redeemer = asRedeemer $ MustMintCurrencyWithRedeemer coinMintingPolicyHashV2 unitRedeemer tknName tknAmount
+        lookups0 = Constraints.plutusV2MintingPolicy mustMintPolicyV2
+                <> Constraints.plutusV2MintingPolicy coinMintingPolicyV2
+        tx0 = Constraints.mustMintCurrencyWithRedeemer mustMintPolicyHashV2 redeemer tknName 1
+           <> Constraints.mustMintCurrencyWithRedeemer coinMintingPolicyHashV2 unitRedeemer tknName tknAmount
```

Comment:
Seems like all of these lines are not necessary, afaiu. For the 1st transaction, you simply need to create an UTXO which contains an inlined minting script.